cruise-control/src/integrationTest/java/com/linkedin/kafka/cruisecontrol/DiskFailureIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+  
+  @Before
+  public void setup() throws Exception {
+    super.start();
+  }
+
+  @Override
+  protected int clusterSize() {
+    return KAFKA_CLUSTER_SIZE;
+  }
+
+  @After
+  public void teardown() {
+    super.stop();
+  }
+
+  @Override
+  public Map<Object, Object> overridingProps() {
+    Map<Object, Object> props = KafkaCruiseControlIntegrationTestUtils.createBrokerProps();
+    Entry<File, File> logFolders = Map.entry(CCKafkaTestUtils.newTempDir(), CCKafkaTestUtils.newTempDir());
+    _brokerLogDirs.add(logFolders);
+    props.put(KafkaConfig.LogDirsProp(), logFolders.getKey().getAbsolutePath() + "," + logFolders.getValue().getAbsolutePath());
+    return props;
+  }
+
+  @Override
+  protected Map<String, Object> withConfigs() {
+    Map<String, Object> configs = KafkaCruiseControlIntegrationTestUtils.ccConfigOverrides();
+    configs.put(TopicReplicationFactorAnomalyFinder.SELF_HEALING_TARGET_TOPIC_REPLICATION_FACTOR_CONFIG, "2");
+    
+    configs.put(AnalyzerConfig.DEFAULT_GOALS_CONFIG, new StringJoiner(",")
+        .add(ReplicaDistributionGoal.class.getName())
+        .add(ReplicaCapacityGoal.class.getName())
+        .add(DiskCapacityGoal.class.getName()).toString());
+    
+    configs.put(AnalyzerConfig.HARD_GOALS_CONFIG, new StringJoiner(",")
+        .add(ReplicaCapacityGoal.class.getName())
+        .add(DiskCapacityGoal.class.getName()).toString());
+    
+    return configs;
+  }
+
+  @Test
+  public void testDiskFailure() throws IOException {
+    AdminClient adminClient = KafkaCruiseControlUtils.createAdminClient(Collections
+        .singletonMap(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, broker(0).plaintextAddr()));
+    try {
+      adminClient.createTopics(Arrays.asList(new NewTopic(TOPIC0, PARTITION_COUNT, TOPIC0_REPLICATION_FACTOR)));
+    } finally {
+      KafkaCruiseControlUtils.closeAdminClientWithTimeout(adminClient);
+    }
+
+    waitForMetadataPropogates();
+
+    KafkaCruiseControlIntegrationTestUtils.produceRandomDataToTopic(TOPIC0, 5, 400, KafkaCruiseControlIntegrationTestUtils
+        .getDefaultProducerProperties(bootstrapServers()));
+
+    waitForProposal();
+    
+    Entry<File, File> entry = _brokerLogDirs.get(BROKER_ID_TO_CAUSE_DISK_FAILURE);
+    FileUtils.deleteDirectory(entry.getKey());
+    LOG.info("Disk failure injected");
+    waitForOfflineReplicaDetection();
+    waitForDiskFailureFixStarted();
+    waitForOfflineReplicasFixed();
+  }
+
+  private void waitForOfflineReplicasFixed() {
+    KafkaCruiseControlIntegrationTestUtils.waitForConditionMeet(() -> {
+      String responseMessage = KafkaCruiseControlIntegrationTestUtils
+          .callCruiseControl(_app.serverUrl(), CRUISE_CONTROL_KAFKA_CLUSTER_STATE_ENDPOINT);
+      Map<String, String> offlineReplicas = JsonPath.read(responseMessage,
+            "$.KafkaBrokerState.OfflineReplicaCountByBrokerId");
+      return offlineReplicas.isEmpty();
+    }, 800, new AssertionError("There are still offline replica on broker"));
```

Comment:
(Applicable to the similar timeouts in integration tests) Why do we expect such large timeouts? -- i.e. what is taking so much time in these tests?

Due to integration tests taking too long, we have been observing test failures in our CI -- i.e. `Too long with no output (exceeded 10m0s): context deadline exceeded`. As a result, a recent PR had to bump up CI timeout as a temporary workaround (see https://github.com/linkedin/cruise-control/pull/1749).

Can we make sure that all aggregate timeouts (including retries) in integration tests is below `10 minutes`?

Reply:
Hi @efeg,
An integration test execution time is about two and three minutes. With four tests the ITs run time between 8 and 12 mins. I can decrease the timeouts but the tests execution time can be over 10 mins easily. 
We can enable the tests logging to avoid no output issue during the ITs but I think this is not the best way to fix this. What is your opinion? Do you know any other solution?

Reply:
In my opinion, these tests could be useful for CC so I hope we will find a solution to merge these. :)

Reply:
I think there are two actions to take:
1. We should figure out what is taking time in each integration test, and try to address that. For example, if the timeout is due to goal optimization times, we can configure the corresponding goal parameters to reduce that time.
2. Logging some success output after each integration test may provide a better view of what is going on during the overall integration testing process.

Reply:
Thanks for your ideas. I am going to add minimal logging for ITs like the unit tests have. I reduced the test execution time as much I could.

Reply:
@efeg What gradle task are you using for testing? In a previous commit we created the `integrationTest` task to avoid running together unit tests and integration test. Perhaps we could hook this into a separate task or don't extend the `test` task even to avoid running them during a test phase. Would that work?

Reply:
Hi @efeg 
Did you have time to check the changes?

Reply:
@viktorsomogyi @lkalina-cera 
Sorry for the delayed response! It is a good idea to avoid running integration tests for each PR -- maybe we can run them as part of a separate task only when we publish a new release. What do you think?

Reply:
@lkalina-cera @viktorsomogyi -- quick reminder on this.

Reply:
Thanks for the ping and sorry for the late reply :).
I think running it on a per release basis is good. I had a quick look on the circleci setup and I think the problem is that we're integrating into the `check` task but at least I can unlink it so it won't run when `build` is triggered.
The regarding the per release run: would it be enough if we defined an extra job in the circleci workflow (let's call it `release-test`) and integrated it into the `build-and-publish` workflow?

Reply:
Thanks for the reply Viktor!
Yes, I think the proposed approach should work.

Reply:
@viktorsomogyi Thanks for the commit -- I unsuccessfully tried making some naive changes to run integration-test only during publish. I suspect we might be able to figure out the right config, but didn't get a chance to dig into the documentation -- i.e.  https://circleci.com/docs/2.0/workflows/